pkg/webhooks/pod_webhook.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhooks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	jobset "sigs.k8s.io/jobset/api/jobset/v1alpha2"
+)
+
+// +kubebuilder:webhook:path=/mutate--v1-pod,mutating=true,failurePolicy=fail,groups="",resources=pods,verbs=create;update,versions=v1,name=mpod.kb.io,sideEffects=None,admissionReviewVersions=v1
+
+// podMutatingWebhook for mutating webhook.
+type podMutatingWebhook struct {
+	client  client.Client
+	decoder *admission.Decoder
+}
+
+func NewPodMutatingWebhook(mgr ctrl.Manager) *podMutatingWebhook {
+	return &podMutatingWebhook{client: mgr.GetClient()}
+}
+
+// SetupMutatingWebhook configures the mutating webhook for pods.
+func (p *podMutatingWebhook) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(&corev1.Pod{}).
+		WithDefaulter(p).
+		Complete()
+}
+
+func (p *podMutatingWebhook) Default(ctx context.Context, obj runtime.Object) error {
+	return nil
+}
+
+func (p *podMutatingWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
```

Comment:
this handles admission, not mutations; we need to put this logic in the Default function above